service.py

```diff
@@ -0,0 +1,48 @@
+import threading
+import os
+
+from ml_tools.tools import frame_to_jpg
+
+from pydbus import SystemBus
+from gi.repository import GLib
+
+SNPASHOT_NAME = "still.png"
+DBUS_NAME = "org.cacophony.thermalrecorder"
+DBUS_PATH = "/org/cacophony/thermalrecorder"
+
+
+class Service(object):
+    """
+        <node>
+            <interface name='org.cacophony.thermalrecorder'>
+                <method name='TakeSnapshot'>
+                    <arg type='s' name='response' direction='out'/>
+                </method>
+            </interface>
+        </node>
+    """
+
+    def __init__(self, processor):
+        self.processor = processor
+
+    def TakeSnapshot(self):
+        last_frame = self.processor.get_recent_frame()
```

Comment:
Are the things accessed by `get_recent_frame` thread safe? The snapshot service and main thread could be accessing them concurrently.

Reply:
I am fairly sure they are thread safe, i changed the current index (of the sliding windows) to be updated after inserting into the numpy array, so it should always return a valid entry maybe 1 frame behind sometimes.